Rudd johnson/validate registration

RuddJohnson opened the pull request.

Jstar1 committed:
Adding upload page

Jstar1 committed:
adds ability for user to upload images

Commit:
added error output to registration page

Commit:
added express-validator dependencies

Commit:
added validation checking in registration in index.js

Commit:
resolve merge conflicts

Commit:
add error output to fixed registration page

Commit:
further updates to index.js

Commit:
debugged validation of registration

Commit:
added next() function call to registration valdiation so passport middleware can be reached

Commit:
added rendering for upload.njk

Commit:
Update index.js

commenting correction

danieleynis commented on routes/index.js:
```diff
@@ -23,10 +23,23 @@ module.exports = function(passport){
+		req.session.invalid = invalid;
+		res.render('register.njk');
+	}
+	 return next();
```
Have you had a chance to test to see if this goes to the passport authentication successfully and creates the user?

danieleynis commented on routes/index.js:
```diff
@@ -23,10 +23,23 @@ module.exports = function(passport){
-    failureRedirect: '/register'
-  }));
+  router.post('/register', function(req,res,next) {
+  req.check('email','Not a valid email address').isEmail();
```
This looks like it needs to get shifted to the right to keep consistent with the coding style conventions of the project.

danieleynis commented on routes/index.js:
```diff
@@ -23,10 +23,23 @@ module.exports = function(passport){
+  router.post('/register', function(req,res,next) {
+  req.check('email','Not a valid email address').isEmail();
+	req.check('createpassword','password must be atleast 5 characters').isLength({min:5});
+	req.check('createpassword','bos passwords must match').equals(req.body.reenterpassword);
```
Misspelled word `bos`, I think you meant "both"?

danieleynis commented on routes/index.js:
```diff
@@ -23,10 +23,23 @@ module.exports = function(passport){
-    successRedirect: '/confirmation',
-    failureRedirect: '/register'
-  }));
+  router.post('/register', function(req,res,next) {
```
Minor: the arguments for the function are all clumped together `function(req,res,next)`, perhaps adding spaces will make it look nicer and keep with coding style. For example: `function(req, res, next)`

danieleynis commented:
Overall I think the code looks good, I will test it also when my comments get addressed. Other than that I just have a minor comment about the tab size, I think the text editor you are using may not be picking up the tab size of 2, as well as that all tabs are converted to spaces. I would highly recommend using [vscode ](https://code.visualstudio.com/). This is nothing major but something to think about.

Commit:
addressing reviews

Commit:
addressing feedback

ghost commented:
I fixed "both" and added TODOs in the commenting to better outline what is still in progress

Jstar1 committed:
Adds posting method for form

Jstar1 committed:
Adds jquery to handle files

Added the jquery so files can be loaded from user disk
Made js directory for storing javascript files

Jstar1 committed:
Adds rendering for upload page

Jstar1 committed:
Adds css styling for upload page

Commit:
resolving duplicate upload rendering

Commit:
added image scaling to upload page

Commit:
added name attribute

danieleynis committed:
Updated upload.njk

* Added type="submit" to upload button (required)
* Removed method="POST" (my misunderstanding this is actually not required and was causing issues for some reason)

Commit:
changed naming of password attribute so validation would work

Commit:
fixed bug that prevented passport function from being called in registration validation

Commit:
changing password attribute to match nunjucks file

Commit:
resolved merge conflicts

Commit:
fixing merge conflicts

Commit:
Merge branch 'RuddJohnson/ValidateRegistration' of https://github.com/danieleynis/Hysh into RuddJohnson/ValidateRegistration

Commit:
output modification in registration.njk

Commit:
finished validation for login and registration

Commit:
routing successful registration to confirmation, and successful login to photos

ghost commented:
completed functionality for login and registration validation and outputting validation/authentication errors on  front end

danieleynis commented on routes/index.js:
```diff
@@ -22,11 +22,33 @@ module.exports = function(passport){
+
+	if(errors)
+	{
+		console.log(errors);			
```
Remove debugging info, passwords potentially printed to node console running on server

danieleynis commented on routes/index.js:
```diff
@@ -38,15 +60,36 @@ module.exports = function(passport){
+
+	if(errors)
+	{
+		console.log(errors);			
```
Remove console.log here too because of reason in previous comment

danieleynis commented on routes/index.js:
```diff
@@ -38,15 +60,36 @@ module.exports = function(passport){
+		return next(err);
+	}
+	if(proceed){
+		res.render("photos.njk");	//if username is  present, route to photos
```
Here just `res.redirect('/photos')`
